### PR TITLE
PORTALS-882

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/lib/utils/DownloadSpeedTest.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    getEntity(sessionToken, entityId).then((entity: Entity) => {
+      const fileEntity:FileEntity = entity as FileEntity
+      const fileHandleAssociationList = [
+        {
+          associateObjectId: fileEntity.id,
+          associateObjectType: 'FileEntity',
+          fileHandleId: fileEntity.dataFileHandleId,
+        },
+      ]
+      const request: any = {
```
Comment (leem42) [inline src/lib/utils/DownloadSpeedTest.ts]: Could use BatchFileRequest instead of any for the request

File: src/lib/utils/DownloadSpeedTest.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      getFiles(request, sessionToken)
+        .then((data: BatchFileResult) => {
+          const presignedUrl: string = data.requestedFiles[0].preSignedURL
+          const fileHandle: FileHandle = data.requestedFiles[0].fileHandle
+          // start test!
+          const startMs = new Date().getTime()
+          return getFileHandleContent(fileHandle, presignedUrl).then(
+            () => {
+              // bytes/second
+              console.log(`Transferred ${fileHandle.contentSize/1000000} MB in ${((new Date().getTime() - startMs - ESTIMATED_CORS_TIME_MS)/1000)} seconds`)
```
Comment (leem42) [inline src/lib/utils/DownloadSpeedTest.ts]: Could remove this console.log

File: src/lib/utils/DownloadSpeedTest.ts (hunk 1 above)
Comment (jay-hodgson) [reply src/lib/utils/DownloadSpeedTest.ts]: Done.

File: src/lib/utils/DownloadSpeedTest.ts (hunk 2 above)
Comment (jay-hodgson) [reply src/lib/utils/DownloadSpeedTest.ts]: Useful calculation, especially if you don't trust the reported rate.  So I just commented out.

**Pull request merged**